front/src/components/HomePage/HomePage.js

```diff
@@ -1,24 +1,43 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { connect } from "react-redux";
 import fetchBooks from "../../state/books/actions";
+import Search from "./Search";
 import BookCard from "./BookCard";
 import style from "./HomePage.module.css";
 
-class HomePage extends React.Component {
+const HomePage = (props) => {
+  const { fetchBooks, books } = props;
 
-  componentDidMount() {
-    this.props.fetchBooks()
-  }
+  const [filteredBooks, setFilteredBooks] = useState({
+    filteredTitles: [],
+    filter: "",
+  });
 
-  render() {
-    return (
-      <>
-        <div className={style.bookCards}>
-          {this.props.books.map((book) => (<BookCard book={book} />))}
-        </div>
-      </>
+  const onValueChange = (event) => {
+    let filteredTitles = books.filter((book) =>
+      book.title.toLowerCase().includes(event.target.value.toLowerCase().trim())
     );
-  }
+    setFilteredBooks({ filteredTitles, filter: event.target.value });
+  };
+
+  useEffect(() => {
+    fetchBooks();
+  }, []);
+
+  return (
+    <>
+      <Search onValueChange={onValueChange} />
+      <div className={style.bookCards}>
+        {filteredBooks.filter === "" ? (
+          books.map((book) => <BookCard book={book} />)
+        ) : filteredBooks.filteredTitles.length === 0 ? (
+          <div style={{fontSize: 20}}>Niestety w naszej ofercie nie mamy takiej książki</div>
+        ) : (
+          filteredBooks.filteredTitles.map((book) => <BookCard book={book} />)
+        )}
+      </div>
+    </>
+  );
```

Comment:
```suggestion
  return (
    <>
      <Search onValueChange={onValueChange} />
      <div className={style.bookCards}>
        {bookList.length === 0 ? (
          <div style={{fontSize: 20}}>Niestety w naszej ofercie nie mamy takiej książki</div>
        ) : (
          bookList.map((book) => <BookCard book={book} />)
        )}
      </div>
    </>
  );
```
That way we can avoid nested conditions, which are hard to ready for me